webapi/src/main/java/org/sakaiproject/webapi/controllers/SitesController.java

```diff
@@ -85,4 +93,88 @@
 
         return data;
 	}
+
+    @Resource
+    private SecurityService securityService = (SecurityService) ComponentManager.get(SecurityService.class);
+
+    @Resource
+    private PreferencesService preferencesService = (PreferencesService) ComponentManager.get(PreferencesService.class);
+
+    @Resource
+    private ActiveToolManager activeToolManager = (ActiveToolManager) ComponentManager.get(ActiveToolManager.class);
+
+    @Resource
+    private ToolManager toolManager = (ToolManager) ComponentManager.get(ToolManager.class.getName());
+
+	@GetMapping(value = "/users/{userId}/favs", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, List<Map<String, Object>>> getFavsWithPages(@PathVariable String userId)
+        throws UserNotDefinedException {
+
+        Session session = checkSakaiSession();
+        String reqUser = session.getUserId();
+        //If non admin user requests data of another user, return empty map and warn
+        if(!securityService.isSuperUser() && !userId.equals(reqUser)) {
+            log.warn("User " + reqUser + " requested favorites from user " + userId);
+            return new HashMap<>();
+        }
+        return getUserFavoriteSitesWithPages(userId);
+	}
+
+	@GetMapping(value = "/user/favs", produces = MediaType.APPLICATION_JSON_VALUE)
+    public Map<String, List<Map<String, Object>>> getFavsWithPages()
+        throws UserNotDefinedException {
+
+        Session session = checkSakaiSession();
+        String reqUser = session.getUserId();
+        return getUserFavoriteSitesWithPages(reqUser);
+	}
+
+    private Map<String, List<Map<String, Object>>> getUserFavoriteSitesWithPages(String userId)
+        throws UserNotDefinedException  {
+
+        ResourceProperties props = preferencesService.getPreferences(userId).getProperties(org.sakaiproject.user.api.PreferencesService.SITENAV_PREFS_KEY);
```

Comment:
resourceProperties instead of props

Reply:
I think preferencesService.getPreferences(userId) could be nullable for new users?

Reply:
getPreferences can't be null, but getPropertyList, I'll return an empty map in that case